src/components/App.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  CssBaseline,
+  createStyles
+} from '@material-ui/core';
+import { FolderList } from './FolderList';
+import { useFolderList } from '../hooks/useFolderList';
+
+export const styles = createStyles({
```

Comment:
so it's best best practice in a `tsx` file to have three langages?

* css
* html
* typescript

:interrobang: 

i find this strange from a _clean code_ approach, but i'll have your word for it if you say: 

> yep, that's the way to go with this kind of file!

Reply:
This is explained here https://vuejs.org/v2/guide/single-file-components.html#What-About-Separation-of-Concerns

It is from vuejs documentation and we are using react, but the approach is the same

Reply:
I won't fight the framework.

It is an _interesting_ idea that has the benefit to be conscious and taken seriously from what is provided by the provided link.

I will stay open minded and will hope that this is an acquired taste.

Thank you for the link.